certbot/src/certbot/_internal/renewal.py

```diff
@@ -325,6 +326,51 @@ def should_renew(config: configuration.NamespaceConfig,
     return False
 
 
+def _ari_renewal_time(config: configuration.NamespaceConfig,
+                      lineage: storage.RenewableCert,
+                      acme_clients: Dict[str, acme_client.ClientV2],
+                      cert_pem: bytes,
+                      )-> Optional[datetime.datetime]:
+    """Return the ARI suggested renewal time if it's available."""
+    # For ARI requests, we want to use the ACME directory URL from which the
+    # cert was originally requested. Since `config.server` can be overridden on
+    # the command line, we're using the server stored in the cert's renewal
+    # conf, i.e. `lineage.server`
+    #
+    # Fixes https://github.com/certbot/certbot/issues/10339
+    if lineage.server:
+        try:
+            # Creating a new ACME client makes a network request, so check if we have
+            # one cached for this cert's server already
+            if lineage.server not in acme_clients:
+                acme_clients[lineage.server] = \
+                    client.create_acme_client(config, server_override=lineage.server)
+            acme = acme_clients.get(lineage.server, None)
+
+            # Attempt to get the ARI-defined renewal time
+            if acme:
+                return acme.renewal_time(cert_pem)[0]
+        except Exception as exception:  # pylint: disable=broad-except
+            # We want to stop errors around ARI preventing renewal so we catch all exceptions here
+            # with a warning asking users to tell us about any problems they are experiencing
+            logger.warning("An error occurred requesting ACME Renewal Information (ARI). If this "
+                           "problem persists and you think it's a bug in Certbot, please open an "
+                           "issue at https://github.com/certbot/certbot/issues/new/choose.")
+            # While not strictly necessary, we unwrap any ARIErrors here to create slightly
```

Comment:
I agree that it's cleaner to not print the intermediate error. But I worry that the underlying error will not be sufficiently clear and explain to the user what happened. Printing both lets us attempt to explain to the user where things went wrong. Seems like if we do receive a report of an error, having that additional context would make things easier for us to debug.